src/main/java/de/hysky/skyblocker/skyblock/dwarven/CarpetHighlighter.java

```diff
@@ -0,0 +1,103 @@
+package de.hysky.skyblocker.skyblock.dwarven;
+
+import de.hysky.skyblocker.annotations.Init;
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.events.SkyblockEvents;
+import de.hysky.skyblocker.utils.Boxes;
+import de.hysky.skyblocker.utils.Location;
+import de.hysky.skyblocker.utils.Resettable;
+import de.hysky.skyblocker.utils.render.RenderHelper;
+import de.hysky.skyblocker.utils.render.Renderable;
+import de.hysky.skyblocker.utils.scheduler.Scheduler;
+import it.unimi.dsi.fastutil.objects.ObjectAVLTreeSet;
+import net.fabricmc.fabric.api.client.networking.v1.ClientPlayConnectionEvents;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderContext;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.CarpetBlock;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+
+import java.awt.*;
+
+/**
+ * Highlights unbreakable carpets within ore veins in the Dwarven Mines.
+ */
+public final class CarpetHighlighter implements Renderable, Resettable {
+	public static final CarpetHighlighter INSTANCE = new CarpetHighlighter();
+
+	private static final Vec3d CARPET_BOUNDING_BOX = Boxes.getLengthVec(CarpetBlock.SHAPE.getBoundingBox());
+	private static final int SEARCH_RADIUS = 15;
+	private static final int TICK_INTERVAL = 15;
+	private static final ObjectAVLTreeSet<BlockPos> CARPET_LOCATIONS = new ObjectAVLTreeSet<>();
+	private static float[] colorComponents;
+	private static boolean isLocationValid = false;
+
+	@Init
+	public static void init() {
+		INSTANCE.configCallback(SkyblockerConfigManager.get().mining.dwarvenMines.carpetHighlightColor);
```

Comment:
This retrieves the value once meaning in order for colour changes to be applied the game needs to be restarted, was this intended?

Reply:
The color is decided with a callback from config change to reflect any changes, as the method name `configCallback` implies
https://github.com/Emirlol/Skyblocker/blob/0abf3775222fcccdac4c4c57df8867e7cfa05502/src/main/java/de/hysky/skyblocker/config/categories/MiningCategory.java#L74
That line makes it so that the initial value is also applied rather than only updates

Reply:
oh didn't see that.